### PR TITLE
Add sitemap loader

### PR DESCRIPTION
**Pull request by selamanse**
- Adds a a Sitemap Loader which uses the AsyncWebPageReader
- Adds tests for this loader
- Adds Readme for this loader and
- Updates Readme for AsyncWebPageReader

I have locally tested this loader by adjusting the loader_hub_url to this working branch

### CODE REVIEW

File: llama_hub/web/sitemap/README.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+This loader is an asynchronous web scraper that fetches the text from static websites by using its sitemap and optionally converting the HTML to text.
+
+It is based on the [Async Website Loader](../async_web/README.md)
+
+## Usage
+
+To use this loader, you just declare the sitemap.xml url like this:
+
+```python
+from llama_index import download_loader
```
Comment (jerryjliu) [inline llama_hub/web/sitemap/README.md:12]: slight nit: we changed default usage (llama-hub is a pypi package) so it should now be 

```python
from llama_hub.web.sitemap.base import AsyncWebPageReader
```

File: llama_hub/web/sitemap/README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Sitemap Loader
+
+This loader is an asynchronous web scraper that fetches the text from static websites by using its sitemap and optionally converting the HTML to text.
+
+It is based on the [Async Website Loader](../async_web/README.md)
```
Comment (jerryjliu) [inline llama_hub/web/sitemap/README.md:5]: this link won't render on llamahub - could you provide the absolute link to the llamahub repo https://llama-hub-ui.vercel.app/l/web-async_web

**Review by jerryjliu — COMMENTED** [review]
looks pretty good @selamanse! some minor comments

File: llama_hub/web/sitemap/README.md (hunk 1 above)
Comment (selamanse) [reply llama_hub/web/sitemap/README.md]: done 👍🏻

File: llama_hub/web/sitemap/README.md (hunk 2 above)
Comment (selamanse) [reply llama_hub/web/sitemap/README.md]: thx for pointing out. fixed it 👍🏻

**Review by jerryjliu — APPROVED** [review]

File: llama_hub/web/sitemap/README.md (hunk 1 above)
Comment (selamanse) [reply llama_hub/web/sitemap/README.md]: I updated the readme again, I think I had some errors in the example: 

https://github.com/emptycrown/llama-hub/pull/336

**Pull request merged**